### PR TITLE
Add the cheesecake factory spider

### LINKED ISSUE #822: Cheesecake Factory
http://locations.thecheesecakefactory.com/

### PR DESCRIPTION
**Pull request by tonyskapunk**
Uses sitemapspider and structreddataspider to crawl and parse locations

Fixes #822

### CODE REVIEW

File: locations/spiders/thecheesecakefactory.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+from scrapy.spiders import SitemapSpider
+
+from locations.hours import OpeningHours
+from locations.structured_data_spider import StructuredDataSpider
+
+
+class UrbanOutfitters(SitemapSpider, StructuredDataSpider):
```
Comment (mjoe999) [inline locations/spiders/thecheesecakefactory.py:8]: Hi Tony, One small comment would be to change the class name to TheCheesecakeFactorySpider. Cheers, Joe.
PS. The StructuredDataSpider is pretty cool I think good to see you're having some "fun" with it.

Comment (tonyskapunk) [reply locations/spiders/thecheesecakefactory.py]: Thanks for reviewing @mjoe999; it is now fixed.

Regarding the StructureDataSpider, it's excellent to reduce repeating ourselves so much.

File: locations/spiders/thecheesecakefactory.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+class TheCheesecakeFactorySpider(SitemapSpider, StructuredDataSpider):
+    download_delay = 0.1
+    name = "thecheesecakefactory"
+    item_attributes = {"brand": "The Cheesecake Factory", "brand_wikidata": "Q1045842"}
+    allowed_domains = [
+        "www.thecheesecakefactory.com",
+        "locations.thecheesecakefactory.com",
+    ]
+    sitemap_urls = ["https://locations.thecheesecakefactory.com/robots.txt"]
+    sitemap_rules = [("", "parse_sd")]
```
Comment (Cj-Malone) [inline locations/spiders/thecheesecakefactory.py:17]: If we checkout the sitemap, it includes pages for the [states](https://locations.thecheesecakefactory.com/ok/) and the [cities](https://locations.thecheesecakefactory.com/ok/tulsa/), we don't need to fetch them, the following filter only matches on the store pages.

```suggestion
    sitemap_rules = [(r"https:\/\/locations\.thecheesecakefactory\.com\/.+\.html", "parse_sd")]
```

**Review by Cj-Malone — APPROVED** [review]
LGTM, thank you.

File: locations/spiders/thecheesecakefactory.py (hunk 2 above)
Comment (tonyskapunk) [reply locations/spiders/thecheesecakefactory.py:17]: Thanks I've added it in #4095

**Pull request merged**